embulk-core/src/main/java/org/embulk/spi/ExecSession.java

```diff
@@ -207,6 +211,16 @@ public TimestampFormatter newTimestampFormatter(String format, DateTimeZone time
         return new TimestampFormatter(format, formatterTask);
     }
 
+    public void setReporters(final Map<Reporter.Channel, Reporter> reporters)
+    {
+        this.reporters = reporters;
```

Comment:
- Will rename this method with setReportersUnsafe
- Add javadoc comment about why this method needs to have public